zallet/src/components/json_rpc/server/error.rs

```diff
@@ -51,6 +51,40 @@ pub enum LegacyCode {
     ClientNodeNotConnected = -29,
     /// Invalid IP/Subnet
     ClientInvalidIpOrSubnet = -30,
+
+    // Wallet errors
+    /// Unspecified problem with wallet (key not found etc.)
+    Wallet = -4,
+    /// Not enough funds in wallet or account
+    WalletInsufficientFunds = -6,
+    /// Accounts are unsupported
+    WalletAccountsUnsupported = -11,
+    /// Keypool ran out, call keypoolrefill first
+    WalletKeypoolRanOut = -12,
+    /// Enter the wallet passphrase with walletpassphrase first
+    WalletUnlockNeeded = -13,
+    /// The wallet passphrase entered was incorrect
+    WalletPassphraseIncorrect = -14,
+    /// Command given in wrong wallet encryption state (encrypting an encrypted wallet etc.)
+    WalletWrongEncState = -15,
+    /// Failed to encrypt the wallet
+    WalletEncryptionFailed = -16,
+    /// Wallet is already unlocked
+    WalletAlreadyUnlocked = -17,
+    /// User must acknowledge backup of the mnemonic seed.
+    WalletBackupRequired = -18,
+}
```

Comment:
This doesn't have the standard JSON-RPC 2.0 errors:

https://github.com/zcash/zcash/blob/a3435336b0c561799ac6805a27993eca3f9656df/src/rpc/protocol.h#L35-L40

Reply:
Those are provided by `jsonrpsee` IIRC